File: submissions/1254-deepest-leaves-sum/solution.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

function deepestLeavesSum(root: TreeNode | null): number {
  if(!root){
     return 0;
  }
  
  let ans:number = 0;
  let queue = [root]
  
  while(queue.length){
     let lengthOfCurrentLevel = queue.length;
     let nextQueue = [];
      
     for (let i = 0; i < lengthOfCurrentLevel; i++) {
         
         let node = queue[i];
         
         if(node.left){
            nextQueue.push(node.left) 
         }
         if(node.right){
             nextQueue.push(node.right)
         } else if(!nextQueue.length){
             ans = queue.reduce((acc, el)=>acc+=el.val, 0);
         }
     }
     queue = nextQueue 
  }

    return ans;
};

File: submissions/1254-deepest-leaves-sum/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

from collections import deque

class Solution:
    def deepestLeavesSum(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0
        
        queue = deque([root])
        last_level = []
        
        while queue:
            curr_len = len(queue)
            last_level = []
            
            for _ in range(curr_len):
                node = queue.popleft()
                last_level.append(node.val) 

                if node.left:
                    queue.append(node.left)
                if node.right:
                    queue.append(node.right)
                    
        return sum(last_level)  
                
